File: src/main/java/me/pljr/pljrapispigot/utils/ScoreboardUtil.java
package me.pljr.pljrapispigot.utils;

import me.pljr.pljrapispigot.builders.ScoreboardBuilder;
import me.pljr.pljrapispigot.objects.PLJRScoreboard;
import org.bukkit.Bukkit;
import org.bukkit.entity.Player;

import java.util.ArrayList;
import java.util.List;

public final class ScoreboardUtil {
    /**
     * Sends {@link PLJRScoreboard} to {@link Player}
     *
     * @param player {@link Player} that should receive the scoreboard
     * @param scoreboard {@link PLJRScoreboard} that should be sent to player
     */
    public static void send(Player player, PLJRScoreboard scoreboard){
        ScoreboardBuilder builder = new ScoreboardBuilder(scoreboard);
        List<String> linesWithPAPI = new ArrayList<>();
        for (String line : builder.getLines()){
            linesWithPAPI.add(PapiUtil.setPlaceholders(player, line));
        }
        builder.withLines(linesWithPAPI);
        player.setScoreboard(builder.create().parseScoreboard());
    }

    /**
     * Broadcasts {@link PLJRScoreboard} to all players.
     *
     * @param scoreboard {@link PLJRScoreboard} that should be broadcasted
     *
     * @see #send(Player, PLJRScoreboard)
     */
    public static void broadcast(PLJRScoreboard scoreboard){
        for (Player player : Bukkit.getOnlinePlayers()){
            send(player, scoreboard);
        }
    }
}

File: src/main/kotlin/me/pljr/pljrapispigot/scoreboard/ScoreboardUtil.kt
package me.pljr.pljrapispigot.scoreboard

import me.pljr.pljrapispigot.util.setPlaceholders
import org.bukkit.Bukkit
import org.bukkit.entity.Player

/**
 * Sends [PLJRScoreboard] to [Player]
 *
 * @param player [Player] that should receive the scoreboard
 * @param scoreboard [PLJRScoreboard] that should be sent to player
 */
fun sendScoreBoard(player: Player, scoreboard: PLJRScoreboard) {
    val builder = ScoreboardBuilder(scoreboard)
    val linesWithPAPI: MutableList<String> = ArrayList()
    builder.lines.forEach { linesWithPAPI.add(setPlaceholders(player, it)) }
    builder.withLines(linesWithPAPI)
    player.scoreboard = builder.create().toScoreboard()
}

/**
 * Broadcasts [PLJRScoreboard] to all players.
 *
 * @param scoreboard [PLJRScoreboard] that should be broadcasted
 */
fun broadcastScoreBoard(scoreboard: PLJRScoreboard) {
    Bukkit.getOnlinePlayers().forEach { sendScoreBoard(it, scoreboard) }
}